### PR TITLE
Add transfer learning model builder sub-module

### PR DESCRIPTION
**Pull request by laksh22**
# Description

Users may want to conduct transfer learning on a pre-trained model. For this purpose, `tf.keras.applications` can be used. A mode builder has been added which allows the user to choose the model they want to train on, add/edit/delete additional layers,and  choose the epochs, batch size, learning rate, loss, optimizer, and metric. 

> Note: Since the preprocessing steps sub-module and the model builder sub-module use the same modal for adding layers/steps, a single component called `AddModelEntityModal` has been created.
> An option for choosing the source type has been added when a user created a new model. This is because a user will have a different UI based on if a model is being built from scratch, from transfer learning, or from an uploaded model.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

1. Created a new model
2. Added 2 new layers
3. Edited both of the layers
4. Deleted both of the layers
5. Changed the epochs, batch size, learning rate, loss, optimizer, and metrics

**Test Configuration**:

Empty sub-module:
p1
Loss options:
p2
Optimizer options:
p3
Metric options:
p4
Model options:
p5
Layer options:
p6
Model with layers:
p7

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by laksh22** [comment]
@ivantha Let me know if any changes are required :)
